pkg/ruler/manager.go

```diff
@@ -111,6 +106,11 @@ func (r *DefaultMultiTenantManager) SyncRuleGroups(ctx context.Context, ruleGrou
 		r.syncRulesToManager(ctx, userID, ruleGroup)
 	}
 
+	// A lock is taken to ensure if this function is called concurrently, then each call
+	// returns after the call map files and check for updates
```

Comment:
This comment is no longer true and should be removed.

However, the most important question is: do we need to enforce two `SyncRuleGroups()` invocations can't run concurrently? Can it ever happen? I checked Mimir and GEM code and I think it **can't** happen. If @gotjosh confirms, I would just add a comment to the `SyncRuleGroups()` function saying it shouldn't be called concurrently.

Reply:
Discussed this with @pracucci offline, I thought we needed this initially but turns out we don't as Marco proved me otherwise.

Within `run()` a trigger of `SyncRuleGroups` can't happen concurrently due to a membership change or periodic timeout which I believe renders this comment incorrect.